Symfony/CS/Tokenizer/Tokens.php

```diff
@@ -929,6 +929,28 @@ public function isArrayMultiLine($index)
     }
 
     /**
+     * Check if there is an anonymous class under given index.
+     *
+     * @param int $index
+     *
+     * @return bool
+     */
+    public function isAnonymousClass($index)
+    {
+        $token = $this[$index];
+
+        if (!$token->isClassy()) {
+            throw new \LogicException('No classy token at given index');
```

Comment:
we want to return `false` instead of throwing an exception. So this should be possible without checking token kind before:

```php
if ($tokens->isAnonymousClass($anyTokenIndex) {}
```